src/main/java/htsjdk/samtools/BAMFileSpan.java

```diff
@@ -115,15 +115,55 @@ public SAMFileSpan removeContentsBefore(final SAMFileSpan fileSpan) {
         validateSorted();
 
         final BAMFileSpan trimmedChunkList = new BAMFileSpan();
+        long chunkStart = bamFileSpan.chunks.get(0).getChunkStart();
         for(final Chunk chunkToTrim: chunks) {
-            if(chunkToTrim.getChunkEnd() > chunkToTrim.getChunkStart()) {
-                if(chunkToTrim.getChunkStart() >= bamFileSpan.chunks.get(0).getChunkStart()) {
+            if(chunkToTrim.getChunkEnd() > chunkStart) {
```

Comment:
Yes, this was intentional as the previous behaviour was incorrect. If a chunk entirely preceded the cutoff, it would never be removed.

There were no tests for this before, and the method is not used internally by htsjdk, which is possibly how this was missed. I've now added unit tests.